src/osl.imageio/oslinput.cpp

```diff
@@ -140,7 +140,13 @@ class OSLInput : public ImageInput {
 // Obligatory material to make this a recognizeable imageio plugin:
 OIIO_PLUGIN_EXPORTS_BEGIN
 
+#if OIIO_PLUGIN_VERSION >= 22
+OIIO_EXPORT ImageInput::unique_ptr osl_input_imageio_create () {
```

Comment:
Because it needs the deleter.

Reply:
The deleter could be another public symbol of the plugin (who's signature matches the deleter the unique_ptr type already expects).

It just seems cleaner to have a plain `ImageInput*` travel back and forth through the C linkage boundary.

Either way, `sizeof(unique_ptr) == 16` in this case since it has to hang on the function pointer.

Reply:
I could leave the input_imageio_create() functions as they were, but it would require every plugin also supplying a input_imageio_destroy() function.

Reply:
OK, I can give that a try. 

It's too bad we can't actually use C++ properly for functions callable in a dynamic library.